### ISSUE TITLE
Scheduling inside Pools is broken

### ISSUE THREAD

**Issue by halorgium** [celluloid/celluloid#193]
From celluloid/celluloid-io#45, @stakach writes: 

> When you call the pool method on Celluloid::IO actors, messages are not sent to the actors even when they are waiting on IO and hence idle. Basically they act like regular Celluloid actors when using pool.
> 
> Is there any way to detect when a Celluloid::IO actor is waiting on IO completion and hence idle? Currently I have created my own Actor pool and round-robin send messages to them which is a little naive
> 
> {...}
> 
> After some playing around with celluloid-promise I noticed regular actors work the same way.
> 
> Basically a thread can run multiple tasks simultaneously using Fibers, lets say you async 10 tasks to an actor and each sleeps at some point, as they sleep the next task is started in a new Fiber.
> 
> Pool does not schedule tasks to actors if they are executing a task that is sleeping and instead waits for an actor to completely finish a task before scheduling again.
> 
> This behaviour makes sense as a single actor in the pool may end up juggling a bunch of fibers while other actors in the pool are doing nothing. Let me know if I am incorrect here.
> 
> {...}
> 
> https://gist.github.com/stakach/5204160
>
> "single executed in   0.000000   0.000000   0.000000 (  0.000856)\n"
> "pool executed in   0.010000   0.000000   0.010000 ( 10.007874)\n"
> D, [2013-03-20T23:06:19.005524 #2809] DEBUG -- : Terminating 8 actors...
> D, [2013-03-20T23:06:19.007185 #2809] DEBUG -- : Shutdown completed cleanly

The bug seems to be present in all `Celluloid` pools.

**Comment by jonhyman** [comment]
We're using a celluloid pool inside a Sidekiq worker to make some web requests using celluloid-io. In our job, we explicitly call `pool.terminate` (even though the docs say that will happen when it gets GC'd). We're seeing what appears to be deadlocking, the call to `terminate` is hanging indefinitely until it is killed by sidekiq. Might this issue be the cause? Has anyone else on this thread experienced deadlocks from pools?

**Comment by digitalextremist** [comment]
Please raise this as a new issue at (celluloid-pool)[/celluloid/celluloid-pool] and I'll look into it. And please test this with `0.17.0` also. The entire `Pool` system was overhauled.

**Comment by jonhyman** [comment]
Okay will do once we upgrade sidekiq to use 0.17.0

**Comment by digitalextremist** [comment]
Right on. Shouldn't be much trouble. Are you talking about patching to use `0.17.0-current` or `0.17.0-backported`?

**Comment by jonhyman** [comment]
Not sure. We're waiting on https://github.com/mperham/sidekiq/issues/2420

**Comment by digitalextremist** [comment]
That should be resolved at mperham/sidekiq#2438
